action/generate.go

```diff
@@ -10,18 +10,22 @@ import (
 	"github.com/justwatchcom/gopass/store/sub"
 	"github.com/justwatchcom/gopass/utils/ctxutil"
 	"github.com/justwatchcom/gopass/utils/pwgen"
+	"github.com/martinhoefling/goxkcdpwgen/xkcdpwgen"
 	"github.com/urfave/cli"
 )
 
 const (
-	defaultLength = 24
+	defaultLength     = 24
+	defaultXKCDLength = 4
 )
 
 // Generate & save a password
 func (s *Action) Generate(ctx context.Context, c *cli.Context) error {
 	force := c.Bool("force")
 	edit := c.Bool("edit")
 	symbols := c.Bool("symbols")
+	xkcd := c.Bool("xkcd")
+	xkcdo := c.Bool("xkcdo")
```

Comment:
This could use a better variable name. I am not sure what distinguishes it from `xkcd` above.

Reply:
Looking further down, I see it causes the separator to be a space. I would instead prefer if you offered an argument for the separator. Perhaps, if you're feeling up to it, a list of separators to be randomly used.

e.g., `--xkcd --separators=",-_"` to randomly select one of comma, dash and underscore as a separator.

Reply:
These variable names are indeed confusing. I was rather thinking about an switch like `delimiter` or `separator` which contains the delimiter. You could also support multiple separators like @gburtini suggested. I'm not sure if that is a good idea, because having to remember the exact placement of each different separator makes the generated password much harder to remember, but we could offer it for users who want it.

Reply:
You're right - I think it disproportionately impacts memorability. Perhaps we should _not_ add it.

It does not add significant bits of entropy [1] if you can assume users are only going to use natural separators.

[1] 4 words from Martin's 7777 wordset is 7777^4 options. Allowing the ASCII character set of separators would add 255^3 more to that set while greatly decreasing memorability.

Reply:
So do you have a strong opinion about a "delimiter" switch? From a user perspective, I see two cases: 
* Easier to type: lowercase + space and 
* Easier to copy / paste 1 Word-> no spaces but readable by inserting caps (you're right - it's not camelcase).

With two explicit switches, I would always need only one switch. With a delim switch, you need two for one of the two cases.

What about these two switches.

xkcdnosep, xn Use multiple random english words combined to a password without spaces/separator. First character of words is capitalized.
xkcdsep, xs Use multiple random english words combined to a password with separator. If no separator is given, the default is space.

The default is chosen if empty string is given in the option.

Reply:
Hmm, unfortunately urfave/cli does not support optional arguments. So I would then go for:

xkcd, x Use multiple random english words combined to a password. If no separator is specified, the words are combined without spaces/separator and the first character of words is capitalised.
xkcdsep, xs Word separator for generated xkcd style password.